components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -1827,6 +1829,32 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         return subscription;
     }
 
+    async getStripePublishableKey(ctx: TraceContext): Promise<string | undefined> {
+        this.checkAndBlockUser("getStripePublishableKey");
+        return this.config.stripeSettings?.publishableKey;
+    }
+
+    async getStripeSetupIntentClientSecret(ctx: TraceContext): Promise<string | undefined> {
```

Comment:
When interactions with 3rd party services fail, we shouldn't be propagating the errors directly to the client. We don't entirely control what is in the errors and may contain sensitive data. How is that handled here? I'd expect that we check for success/error and return appropriate obfuscated error messages upstream to the client.

Reply:
Agree -- we could avoid propagating to the user/client with a try/catch:

```ts
// Pseudo-code
try {
    await stripeThing();
} catch (error) {
    log.error("the full error for our logs", { error });
    throw new Error("Stripe thing failed for some reason ¯\_(ツ)_/¯");
}
```